### PR TITLE
Upgrade to LangChain4j 0.36.2

### PR DESCRIPTION
**Pull request by jmartisk**
- Closes: #1075

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/AiServiceMethodCreateInfo.java [hunk 1]
```diff
@@ -1,7 +1,5 @@
 package io.quarkiverse.langchain4j.runtime.aiservice;
 
-import static org.apache.commons.lang3.StringUtils.EMPTY;
```
Comment (jmartisk) [inline core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/AiServiceMethodCreateInfo.java:3]: This dependency has somehow disappeared, but I don't see why we should re-add it just for a constant of an empty string

File: model-providers/mistral/runtime/src/main/java/io/quarkiverse/langchain4j/mistralai/QuarkusMistralAiClient.java [hunk 2]
```diff
@@ -116,6 +118,12 @@ public MistralAiEmbeddingResponse embedding(MistralAiEmbeddingRequest request) {
         return restApi.embedding(request, apiKey);
     }
 
+    // TODO: we don't provide support for MistralAiModerationModel yet
```
Comment (geoand) [inline model-providers/mistral/runtime/src/main/java/io/quarkiverse/langchain4j/mistralai/QuarkusMistralAiClient.java:121]: We don't?

**Review by geoand — COMMENTED** [review]
Great stuff @jmartisk!

File: model-providers/mistral/runtime/src/main/java/io/quarkiverse/langchain4j/mistralai/QuarkusMistralAiClient.java (hunk 2 above)
Comment (jmartisk) [reply model-providers/mistral/runtime/src/main/java/io/quarkiverse/langchain4j/mistralai/QuarkusMistralAiClient.java:121]: Well, users can instantiate it directly, but we don't have a declarative way via config properties for it yet. I plan to add that via https://github.com/quarkiverse/quarkus-langchain4j/issues/1095

Comment (geoand) [reply model-providers/mistral/runtime/src/main/java/io/quarkiverse/langchain4j/mistralai/QuarkusMistralAiClient.java:121]: Gotcha, thanks

**Pull request merged**